steward/src/somm_send.rs

```diff
@@ -87,14 +89,18 @@ pub async fn data_hash(
     let mut hasher = sha2::Sha256::new();
     if let Some(vote) = &allocation.clone().vote {
         let mut buf = BytesMut::new();
-        vote.encode(&mut buf).unwrap();
+        vote.encode(&mut buf)?;
         let vote_data = hex::encode(&buf);
         let msg = format!("{}:{}:{}", allocation.salt, vote_data, val_address);
         hasher.update(msg.as_bytes());
 
         return Ok(AllocationPrecommit {
             hash: hasher.finalize().to_vec(),
-            cellar_id: vote.cellar.clone().unwrap().id,
+            cellar_id: vote
+                .cellar
+                .clone()
+                .expect("vote contains no 'cellar' value")
```

Comment:
In a function that returns a result, we should probably do that here as well rather than `panic!`.

Reply:
Addressed